code/obj/item/cigarette.dm

```diff
@@ -2,6 +2,136 @@
+				boutput(user, "There's no way you can fit another cigarette in!")
+
+		else
+			//This is just a tricksy way to light the bundle without making this object a subtype of cigarette or copy-paste
+			//Pick a random object from the bundle's contents (should only be cigs sans shenanigans) and call its attackby proc with
+			//the arguments for this current proc. If it lights the cig, call the light proc on this object which lights all cigs.
+			if (islist(contents))
+				var/obj/item/clothing/mask/cigarette/cig = pick(contents)
+				if (istype(cig))
+					cig.attackby(W, user)
+					if (cig.on)
+						src.light(user)
+			else
+				..()
+
+	proc/update_icon()
+		if (islist(contents) && contents.len == 0)
+			return
+
+		//Simple math way you get numbers, almost done, but doesn't look great round((contents.len/max_cigs)*9)
+		var/state_num = 2
+		switch (contents.len)
+			if (2) 			state_num = 2
+			if (3) 			state_num = 3
+			if (4) 			state_num = 4
+			if (5) 			state_num = 5
+			if (6 to 7) 	state_num = 5
+			if (8 to 11) 	state_num = 6
+			if (12 to 15) 	state_num = 7
+			if (16 to 20) 	state_num = 8
+			if (21) 		state_num = 9
+
+
+		if (on)
+			icon_state = "cig_bundle-[state_num]-o"	//9 is the amount of icon states
+		else
+			icon_state = "cig_bundle-[state_num]"
+
+	attack_hand(mob/user as mob)
+		if (islist(contents) && contents.len > 0)
+			var/obj/item/clothing/mask/cigarette/C = pick(contents)
+			contents -= C
+			user.put_in_hand_or_drop(C)
+			boutput(user, "You remove [C] from [src].")
+			// if (C.on && !(C in processing_items))
+			// 	processing_items.Add(C)
+
+
+			//if there's only one cigarette left after we just removed a cig, then just delete this object and put that cig where this is.
+			if (contents.len == 1)
+				var/obj/item/clothing/mask/cigarette/last_cig = contents[1]
+				if (istype(last_cig))
+					// if (last_cig.on && !(last_cig in processing_items))
+					// 	processing_items.Add(last_cig)
+					last_cig.set_loc(user)
+					if (ishuman(user))
+						var/mob/living/carbon/human/H = user
+						if (H.wear_mask == src)
+							H.u_equip(src)
+							H.force_equip(last_cig, H.slot_wear_mask)
+							return
+				qdel(src)
+			update_icon()
+
+
+	dropped(mob/user as mob)
+		src.visible_message("<span style='color:red'>[src] falls apart into a pile of cigarettes!", group = "cig_drop")
+		for (var/obj/item/clothing/mask/cigarette/C in contents)
+			C.set_loc(get_turf(src.loc))
+
+		qdel(src)
+
+	proc/light(var/mob/user as mob)
+		if (src.on == 0)
+			//Silently light all cigs
+			for (var/obj/item/clothing/mask/cigarette/C in contents)
+				C.light(user)
+			src.visible_message("<span style='color:red'>[user] lights all [islist(contents) ? contents.len : "of the"] cigarettes in the bundle!")
+			src.on = 1
+			src.force = 4
+			update_icon()
+
+
+	proc/put_out(var/mob/user as mob, var/message as text)
+		if (src.on == 1)
+			for (var/obj/item/clothing/mask/cigarette/C in contents)
+				C.put_out(user)
+			src.on = -1
```

Comment:
Why `-1` instead of `0`?

Reply:
Cigarettes used -1 to denote a cigarette that's burnt out. That technically shouldn't happen to the bundle since burnt out cigs just fall out, but I just kept it.